internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,491 @@
+	asc.Client = r.Client
+	asc.Object = instance
+	asc.spec = &instance.Spec
+	asc.conditions = &instance.Status.Conditions
+
+	// NOTE: ignoring the Namespace that seems to get set on request when syncing on namespaced objects
+	// when our AgentServiceConfig is ClusterScoped.
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: req.NamespacedName.Name}, instance); err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			return reconcile.Result{}, nil
+		}
+		r.Log.WithError(err).Error("Failed to get resource", req.NamespacedName)
+		return ctrl.Result{}, err
+	}
+
+	if instance.GetDeletionTimestamp().IsZero() {
+		r.Log.Infof("Zero deletion timestamp")
+		// Check to see if the ASC has our finalizer, if so, we must add it
+		if !controllerutil.ContainsFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName) {
+			controllerutil.AddFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName)
+			if err := r.Client.Update(ctx, instance); err != nil {
+				r.Log.WithError(err).Error("failed to add finalizer to AgentServiceConfig")
+			}
+		}
+	} else {
+		r.Log.Infof("Non zero deletion timestamp")
+		err := r.DeleteLocalClusterImportCRs(ctx)
+		if err != nil {
+			r.Log.WithError(err).Error("failed to clean up local cluster CRs")
+			return ctrl.Result{Requeue: true}, err
+		}
+		controllerutil.RemoveFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName)
+		if err := r.Client.Update(ctx, instance); err != nil {
+			r.Log.WithError(err).Error("failed to remove finalizer from AgentServiceConfig")
+			return ctrl.Result{Requeue: true}, err
+		}
+		// Stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.handleManagedCluster(ctx); err != nil {
+		r.Log.WithError(err).Error("error while handling managed cluster")
+		return ctrl.Result{Requeue: true}, err
+	}
+	_ = r.setReconciliationStatus(ctx, instance, true, "LocalClusterImport reconcile completed without error.")
+	// Reconciliation complete
+	return ctrl.Result{}, nil
+}
+
+func (r *LocalClusterImportReconciler) handleManagedCluster(ctx context.Context) error {
+	hasManagedCluster, err := r.HasLocalManagedCluster(ctx)
+	if err != nil {
+		return errors.Wrap(err, "error while attempting to determine presence of managed cluster")
+	}
+	if hasManagedCluster {
+		if err = r.ImportLocalCluster(ctx); err != nil {
+			return errors.Wrap(err, "failed to create managed cluster CRs")
+		}
+		return nil
+	}
+	err = r.DeleteLocalClusterImportCRs(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to clean up local cluster CRs")
+	}
+	return nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *LocalClusterImportReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
```

Comment:
I think you'll want `Owns(...)` for all the objects you're creating so you'll get an event when those are updated or deleted.